src/agent/src/windows_service.cpp

```diff
@@ -195,4 +217,103 @@ namespace WindowsService
         }
     }
 
+    void ServiceStart()
+    {
+        SC_HANDLE hSCManager = OpenSCManager(nullptr, nullptr, SC_MANAGER_CONNECT);
+        if (!hSCManager)
+        {
+            LogError("Error: Unable to open Service Control Manager. Error: {}", GetLastError());
+            return;
+        }
+
+        SC_HANDLE hService = OpenService(hSCManager, AGENT_SERVICENAME.c_str(), SERVICE_START);
+        if (!hService)
+        {
+            LogError("Error: Unable to open service. Error: {}", GetLastError());
+            CloseServiceHandle(hSCManager);
+            return;
+        }
```

Comment:
Up to here, this code is almost the same for `SeriviceStart|ServiceStop|ServiceStatus`. Since we have it 3 times, maybe we can extract a function from this.

Reply:
Done!